node/node_test.go

```diff
@@ -562,6 +562,179 @@ func TestGenerateTransaction_CorrectParamsShouldNotError(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestCreateTransaction_NilAddrConverterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	n, _ := node.NewNode(
+		node.WithMarshalizer(getMarshalizer()),
+		node.WithHasher(getHasher()),
+		node.WithAccountsAdapter(&mock.AccountsStub{}),
+		node.WithTxSignPrivKey(&mock.PrivateKeyStub{}),
+	)
+
+	nonce := uint64(0)
+	value := new(big.Int).SetInt64(10)
+	receiver := ""
+	sender := ""
+	gasPrice := uint64(10)
+	gasLimit := uint64(20)
+	txData := "-"
+	signature := "-"
+	challenge := "-"
+
+	tx, err := n.CreateTransaction(nonce, value, receiver, sender, gasPrice, gasLimit, txData, signature, challenge)
+
+	assert.Nil(t, tx)
+	assert.Equal(t, node.ErrNilAddressConverter, err)
+}
+
+func TestCreateTransaction_NilAccountsAdapterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	n, _ := node.NewNode(
+		node.WithMarshalizer(getMarshalizer()),
+		node.WithHasher(getHasher()),
+		node.WithAddressConverter(&mock.AddressConverterStub{
+			CreateAddressFromHexHandler: func(hexAddress string) (container state.AddressContainer, e error) {
+				return state.NewAddress([]byte(hexAddress)), nil
+			},
+		}),
+		node.WithTxSignPrivKey(&mock.PrivateKeyStub{}),
+	)
+
+	nonce := uint64(0)
+	value := new(big.Int).SetInt64(10)
+	receiver := ""
+	sender := ""
+	gasPrice := uint64(10)
+	gasLimit := uint64(20)
+	txData := "-"
+	signature := "-"
+	challenge := "-"
+
+	tx, err := n.CreateTransaction(nonce, value, receiver, sender, gasPrice, gasLimit, txData, signature, challenge)
+
+	assert.Nil(t, tx)
+	assert.Equal(t, node.ErrNilAccountsAdapter, err)
+}
+
+func TestCreateTransaction_InvalidSignatureShouldErr(t *testing.T) {
+	t.Parallel()
+
+	n, _ := node.NewNode(
+		node.WithMarshalizer(getMarshalizer()),
+		node.WithHasher(getHasher()),
+		node.WithAddressConverter(&mock.AddressConverterStub{
+			CreateAddressFromHexHandler: func(hexAddress string) (container state.AddressContainer, e error) {
+				return state.NewAddress([]byte(hexAddress)), nil
+			},
+		}),
+		node.WithAccountsAdapter(&mock.AccountsStub{}),
+		node.WithTxSignPrivKey(&mock.PrivateKeyStub{}),
+	)
+
+	nonce := uint64(0)
+	value := new(big.Int).SetInt64(10)
+	receiver := "rcv"
+	sender := "snd"
+	gasPrice := uint64(10)
+	gasLimit := uint64(20)
+	txData := "-"
+	signature := "-"
+	challenge := "af4e5"
+
+	tx, err := n.CreateTransaction(nonce, value, receiver, sender, gasPrice, gasLimit, txData, signature, challenge)
+
+	assert.Nil(t, tx)
+	assert.NotNil(t, err)
+}
+
+func TestCreateTransaction_InvalidChallengeShouldErr(t *testing.T) {
```

Comment:
useless test as we do not use tx.challenge

Reply:
removed